verifiers/zksync/src/lib.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2024, Horizen Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub mod benchmarking;
+mod verifier_should;
+mod weight;
+
+use core::marker::PhantomData;
+
+use frame_support::weights::Weight;
+use hp_verifiers::{Cow, Verifier, VerifyError};
+
+pub use native::ZKSYNC_PROOF_SIZE as PROOF_SIZE;
+pub use native::ZKSYNC_PUBS_SIZE as PUBS_SIZE;
+
+#[pallet_verifiers::verifier]
+pub struct Zksync;
+pub use weight::WeightInfo;
+
+impl Verifier for Zksync {
+    type Proof = [u8; PROOF_SIZE];
+
+    type Pubs = [u8; PUBS_SIZE];
+
+    type Vk = ();
+
+    fn hash_context_data() -> &'static [u8] {
+        b"zksync"
+    }
+
+    fn verify_proof(
+        _vk: &Self::Vk,
+        proof: &Self::Proof,
+        pubs: &Self::Pubs,
+    ) -> Result<(), VerifyError> {
+        native::zksync_verify::verify(proof, *pubs).map_err(Into::into)
+    }
+
+    fn pubs_bytes(pubs: &Self::Pubs) -> Cow<[u8]> {
+        Cow::Borrowed(pubs)
+    }
+
+    fn vk_bytes(_vk: &Self::Vk) -> Cow<[u8]> {
+        static VOID: &[u8] = &[];
```

Comment:
Please small comment also here about the rationale

Reply:
Ok